src/rgw/rgw_http_errors.h

```diff
@@ -34,6 +35,8 @@ static inline int rgw_http_error_to_errno(int http_err)
         return -ERR_METHOD_NOT_ALLOWED;
     case 409:
         return -ENOTEMPTY;
+    case 500:
+        return -EAGAIN;
```

Comment:
why EAGAIN instead of ERR_INTERNAL_ERROR? [set_req_state_err()](https://github.com/ceph/ceph/blob/0ed764e/src/rgw/rgw_common.cc#L312-L344) doesn't have a mapping for EAGAIN, so it relies on this:
```c++
  dout(0) << "WARNING: set_req_state_err err_no=" << err_no
        << " resorting to 500" << dendl;

  err.http_ret = 500;
  err.err_code = "UnknownError";
```

in contrast, [rgw_http_s3_errors](https://github.com/ceph/ceph/blob/0ed764e/src/rgw/rgw_common.cc#L127) has a more specific mapping for `{ ERR_INTERNAL_ERROR, {500, "InternalError" }},`

Reply:
changed to `ERR_SERVICE_UNAVAILABLE` and `503`